v2/internal/reconcilers/azure_generic_arm_reconciler.go

```diff
@@ -659,6 +711,39 @@ func (r *AzureDeploymentReconciler) getStatus(ctx context.Context, id string) (g
 	return status, zeroDuration, nil
 }
 
+func (r *AzureDeploymentReconciler) setStatus(status genruntime.ConvertibleStatus) error {
+	// Modifications that impact status have to happen after this because this performs a full
+	// replace of status
+	if status != nil {
+		// SetStatus() takes care of any required conversion to the right version
+		err := r.obj.SetStatus(status)
+		if err != nil {
+			return errors.Wrapf(err, "setting status on %s", r.obj.GetObjectKind().GroupVersionKind())
+		}
+	}
+
+	return nil
+}
+
+func (r *AzureDeploymentReconciler) updateStatus(ctx context.Context) error {
+	resourceID, hasResourceID := genruntime.GetResourceID(r.obj)
+	if !hasResourceID {
+		return errors.Errorf("resource has no resource id")
+	}
+
+	status, _, err := r.getStatus(ctx, resourceID)
+	if err != nil {
+		return errors.Errorf("error getting status for resource ID %q", resourceID)
+	}
+
+	err = r.setStatus(status)
+	if err != nil {
```

Comment:
same here, can avoid extra lines by having one liner assignment and if statement.

Reply:
Done